### PR TITLE
Add adapter: Interlay lending TVL

### PR DESCRIPTION
**Pull request by bvotteler**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Interlay Lending

##### Twitter Link:
https://twitter.com/InterlayHQ

##### Website Link:
http://interlay.io/

##### Logo (High resolution, will be shown with rounded borders):
Same as https://defillama.com/protocol/interlay

##### Current TVL:
approx. $1.67MM

##### Chain:
Interlay

##### Short Description (to be shown on DefiLlama):
Interlay lending total value locked

##### methodology (what is being counted as tvl, how is tvl being calculated):
Sum of tokens supplied (locked) by users for the lending protocol.

##### Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/interlay-dex exports TVL: 
        
 
 
    interlay                  104.66 k
    
    total                    104.66 k

**Comment by llamatester** [comment]
Error while running adapter at projects/interlay-lending: 
        
 
  
    
    
    Error: Data for protocol interlay-lending on chain interlay is undefined on heroku
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/heroku-api.js:11:19
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:63:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:159:9
        at async Promise.all (index 0)
        at async Promise.all (index 0)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:204:3

**Comment by llamatester** [comment]
The adapter at projects/interlay-dex exports TVL: 
        
 
 
    interlay                  104.37 k
    
    total                    104.37 k

**Comment by llamatester** [comment]
Error while running adapter at projects/interlay-lending: 
        
 
  
    
    
    Error: Data for protocol interlay-lending on chain interlay is undefined on heroku
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/heroku-api.js:11:19
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:63:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:159:9
        at async Promise.all (index 0)
        at async Promise.all (index 0)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:204:3

File: projects/interlay-lending/api.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    const [underlyingId, marketData] = data[i];
+
+    const isActive = marketData?.toJSON().state == "Active";
+    if (!isActive) {
+      continue;
+    }
+
+    const lendTokenId = marketData.toJSON().lendTokenId;
+
+    const [exchangeRate, totalIssuance] = Promise.all([
```
Comment (0xngmi) [inline projects/interlay-lending/api.js:23]: you need to separate borrowed from money thats sitting on the contracts

Comment (bvotteler) [reply projects/interlay-lending/api.js:23]: > you need to separate borrowed from money thats sitting on the contracts

Thanks for the feedback.
Just to make sure I understand correctly: Is the requirement to show `locked amounts in lending protocol - current borrowed amounts` ? (ie. only the amounts available for borrowing)

Comment (realdealshaman) [reply projects/interlay-lending/api.js:23]: yes tvl should = total supplied - borrowed and borrowed can be exported separately just as borrowed

**Comment by llamatester** [comment]
The adapter at projects/interlay-dex exports TVL: 
        
 
 
    interlay                  101.28 k
    
    total                    101.28 k

**Comment by llamatester** [comment]
Error while running adapter at projects/interlay-lending: 
        
 
  
    
    
    Error: Data for protocol interlay-lending on chain interlay is undefined on heroku
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/heroku-api.js:11:19
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:63:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:159:9
        at async Promise.all (index 0)
        at async Promise.all (index 0)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:204:3

File: projects/interlay-lending/api.js (hunk 1 above)
Comment (bvotteler) [reply projects/interlay-lending/api.js]: > yes tvl should = total supplied - borrowed and borrowed can be exported separately just as borrowed

Thank you for the confirmation. 
Commit [11db388](https://github.com/DefiLlama/DefiLlama-Adapters/pull/8552/commits/11db388b20d0583bfee6ea44330b7c5db2c64fa0) should fix that.

**Pull request merged**